src/Chapter2.hs

```diff
@@ -858,7 +879,12 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind ls = go ls []
+  where
+    go :: [a] -> [a] -> [a]
+    go [] new_l = new_l
+    go l new_l = new_l ++ [last l] ++ go (init l) new_l
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow to add anything at the end of the list. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
thanks! I think I have a better implementation now:
`    go l new_l = go (tail l) (head l : new_l)`

Reply:
This will fail in runtime on an empty list, as `head` is a partial function.

It is better to use pattern matching in such cases (`[]` and `x:xs`) and avoid using `head` 🙂